wagtail/core/models.py

```diff
@@ -2049,6 +2050,29 @@ def copy(self, recursive=False, to=None, update_attrs=None, copy_revisions=True,
                 page_copy.numchild = numchild
                 page_copy.save(clean=False, update_fields=['numchild'])
 
+            # Copy privacy permission
+            if self.get_view_restrictions().exists() and to:
```

Comment:
`self.get_view_restrictions()` will return all view restrictions that apply to the page, including ones that are defined on a parent or ancestor - we don't want to copy them in that case. `self.view_restrictions.all()` will give the view restrictions defined on this page specifically, so that's what we should use here.

Also, if `to` isn't defined, it means that we're copying the page into the same parent, and in this case we still want to copy the page's restrictions - we shouldn't skip over this block.